src/git.zsh

```diff
@@ -141,27 +141,35 @@ _fzf_complete_git() {
 }
 
 _fzf_complete_git_option_prefix() {
-    if [[ -z ${prefix:/--[^-]##*=*} ]]; then
+    if [[ -z ${prefix:/--[^-]##*=*} ]] || [[ ${prefix:0:2} =~ '-[^-]' ]]; then
         echo ${prefix/=*/=}
     fi
 }
+
 _fzf_complete_git_has_options() {
     local option
     local last_argument=$1
     local prefix=$2
     shift 2
 
     for option in ${(z)@}; do
-        if [[ ${#option} = 1 ]]; then
-            if [[ $last_argument =~ "^-[^-]*$option" ]]; then
+        local option_last_character=${option[-1]}
+        local option_without_last_question_mark=${option%\?}
```

Comment:
These variables are unnecessary because each expression is straightforward enough to understand rather than being assigned to such a lengthy-named variable.